flask_monitoringdashboard/static/elements/modal.html

```diff
@@ -0,0 +1,27 @@
+<div class="modal fade" id="{{ name }}Modal" tabindex="-1" role="dialog"
```

Comment:
Personally, I don't think this is a generalisation worth the effort.

Soon enough you're going to need to want to add an icon to the Save button, customise the styling of the error message, etc... Then you're going to have to either take out this code and paste it back to where it's used or pass a ton of attributes.

Anyway, I'm not saying it's wrong, but I wouldn't have done it. I would have made a live template in my IDE and just generate the modals. Something like this:


![Screen Recording 2020-06-22 at 15 13 55](https://user-images.githubusercontent.com/9981548/85292243-e699ef80-b49b-11ea-8227-0b56de9aa868.gif)